examples/src/chains/conversational_qa_streaming.ts

```diff
@@ -1,52 +1,93 @@
+import { Document } from "langchain/document";
 import { ChatOpenAI } from "langchain/chat_models/openai";
-import { ConversationalRetrievalQAChain } from "langchain/chains";
 import { HNSWLib } from "langchain/vectorstores/hnswlib";
 import { OpenAIEmbeddings } from "langchain/embeddings/openai";
 import { RecursiveCharacterTextSplitter } from "langchain/text_splitter";
-import { BufferMemory } from "langchain/memory";
-
 import * as fs from "fs";
+import { PromptTemplate } from "langchain/prompts";
+import { StringOutputParser } from "langchain/schema/output_parser";
+import { RunnableSequence } from "langchain/schema/runnable";
 
 export const run = async () => {
+  /* Initialize the LLM & set streaming to true */
+  const model = new ChatOpenAI({
+    streaming: true,
+  });
+  /* Load in the file we want to do question answering over */
   const text = fs.readFileSync("state_of_the_union.txt", "utf8");
+  /* Split the text into chunks */
   const textSplitter = new RecursiveCharacterTextSplitter({ chunkSize: 1000 });
   const docs = await textSplitter.createDocuments([text]);
+  /* Create the vectorstore */
   const vectorStore = await HNSWLib.fromDocuments(docs, new OpenAIEmbeddings());
-  let streamedResponse = "";
-  const streamingModel = new ChatOpenAI({
-    streaming: true,
-    callbacks: [
-      {
-        handleLLMNewToken(token) {
-          streamedResponse += token;
-        },
-      },
-    ],
-  });
-  const nonStreamingModel = new ChatOpenAI({});
-  const chain = ConversationalRetrievalQAChain.fromLLM(
-    streamingModel,
-    vectorStore.asRetriever(),
-    {
-      returnSourceDocuments: true,
-      memory: new BufferMemory({
-        memoryKey: "chat_history",
-        inputKey: "question", // The key for the input to the chain
-        outputKey: "text", // The key for the final conversational output of the chain
-        returnMessages: true, // If using with a chat model
-      }),
-      questionGeneratorChainOptions: {
-        llm: nonStreamingModel,
-      },
-    }
+  const retriever = vectorStore.asRetriever();
+
+  const serializeDocs = (docs: Array<Document>) =>
+    docs.map((doc) => doc.pageContent).join("\n\n");
+
+  /**
+   * Create a prompt template for generating an answer based on context and
+   * a question.
+   *
+   * Chat history will be an empty string if it's the first question.
+   *
+   * inputVariables: ["chatHistory", "context", "question"]
+   */
+  const questionPrompt = PromptTemplate.fromTemplate(
+    `Use the following pieces of context to answer the question at the end. If you don't know the answer, just say that you don't know, don't try to make up an answer.\n\nCONTEXT: {context}\n\nCHAT HISTORY: {chatHistory}\n\nQUESTION: {question}\n\nHelpful Answer:`
```

Comment:
Would use actual newlines instead of `\n` for readability here - you can put declare a `const QUESTION_PROMPT_TEMPLATE;` variable above and remove `const run () => ` as it's no longer necessary